pkg/controller/networkpolicy/crd_utils.go

```diff
@@ -131,6 +136,36 @@ func (n *NetworkPolicyController) createAddressGroupForCRD(peer secv1alpha1.Netw
 	return normalizedUID
 }
 
+// createAddressGroupForClusterGroupCRD creates an AddressGroup object corresponding to a
+// ClusterGroup spec. If the AddressGroup already exists, it returns the key
+// otherwise it copies the ClusterGroup CRD contents to an AddressGroup resource and returns
+// its key.
+func (n *NetworkPolicyController) createAddressGroupForClusterGroupCRD(cg *v1alpha2.ClusterGroup) string {
+	// Find the internal Group corresponding to this ClusterGroup
+	igKey := internalGroupKeyFunc(cg)
+	ig, found, _ := n.internalGroupStore.Get(igKey)
+	if !found {
+		klog.V(2).Infof("Internal group %s not found.", igKey)
```

Comment:
Could there be race condition that the internalGroup is created after `createAddressGroupForClusterGroupCRD` since they are all asynchrounous.

Reply:
that is possible.. let me address this in the next commit

Reply:
By next commit, do you mean in this PR or another PR? The case seems not handled yet.

Reply:
another commit in this PR will specifically address this. have not pushed it yet

Reply:
addressed